web-common/tests/utils/start-rill-dev.ts

```diff
@@ -0,0 +1,109 @@
+import { asyncWaitUntil } from "@rilldata/web-common/lib/waitUtils";
+import { getOpenPort } from "@rilldata/web-common/tests/utils/get-open-port";
+import { spawnAndMatch } from "@rilldata/web-common/tests/utils/spawn";
+import axios from "axios";
+import { spawn } from "node:child_process";
+import { cpSync, existsSync, mkdirSync, rmSync } from "node:fs";
+import { join } from "node:path";
+import type { Page } from "playwright";
+import treeKill from "tree-kill";
+
+export const TestTempDirectory = "playwright";
+
+export async function startRillDev(
+  page: Page,
+  use: (page: Page) => Promise<void>,
+  {
+    cliHome,
+    switchEnv,
+    projectDir,
+  }: {
+    cliHome?: string;
+    switchEnv?: boolean;
+    projectDir?: string;
+  },
+) {
+  const TEST_PORT = await getOpenPort();
+  const TEST_PORT_GRPC = await getOpenPort();
+  const TEST_PROJECT_DIRECTORY = join(
+    TestTempDirectory,
+    "projects",
+    "" + TEST_PORT,
+  );
+  // Add a default home if cliHome is not provided so that tests always have a different home than the user's home.
+  // This will make sure that login status won't conflicts with dev's login status when run locally.
+  const TEST_HOME_DIRECTORY = cliHome ?? join(TestTempDirectory, "home");
```

Comment:
Rather than expose the Home directory complexity to all tests via the `cliHome` fixture, can we handle all the logic here? I see both the fallback to "home" here and the "deploy_home" logic elsewhere. Are they intentionally different?

Reply:
We want certain tests to have different logged in user. Especially when testing deploy flow. Happy to help you understand all the different use cases, these were mocked up last week.